### PR TITLE
Pylint checks should be way faster now

### PR DESCRIPTION
**Pull request by potiuk**
Instead of running separate pylint checks for tests and main source
we are running a single check now. This is possible thanks to a
nice hack - we have pylint plugin that injects the right
"# pylint: disable=" comment for all test files while reading
the file content by astroid (just before tokenization)

Thanks to that we can also separate out pylint checks
to a separate job in CI - this way all pylint checks will
be run in parallel to all other checks effectively halving
the time needed to get the static check feedback and potentially
canceling other jobs much faster.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

File: tests/dags/test_logging_in_dag.py [hunk 1]
```diff
@@ -25,6 +25,11 @@
 
 
 def test_logging_fn(**kwargs):
+    """
+    Tests DAG logging.
+    :param kwargs:
+    :return:
+    """
```
Comment (turbaszek) [inline tests/dags/test_logging_in_dag.py:32]: Is this related?

File: tests/airflow_pylint/disable_checks_for_tests.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        elif decoded_lines[0].startswith("#") or decoded_lines[0].strip() == "":
+            decoded_lines[0] = "# pylint: disable=" + DISABLED_CHECKS_FOR_TESTS
+        else:
+            raise Exception(f"The first line of module {mod.name} is not a comment or empty. "
+                            f"Please make sure it is!")
+        # pylint will read from `.file_bytes` attribute later when tokenization
+        mod.file_bytes = "\n".join(decoded_lines).encode("utf-8")
+
+
+MANAGER.register_transform(scoped_nodes.Module, transform)
```
Comment (turbaszek) [inline tests/airflow_pylint/disable_checks_for_tests.py:60]: I'm just wondering if we can first run pylint for main sources and then, just add additional disables to `pylintrc`? Not sure if this will be simpler

Comment (potiuk) [reply tests/airflow_pylint/disable_checks_for_tests.py:60]: This is exactly what we did before and that was a pain.  It took much more  time than this one.
This is a huge optimization, that's why I did it.

The problem is that when you run pylint for tests, it also pulls in all the tested code which means that basically a lot of the main airflow code was parsed and processed twice - once in the "pylint main" and once in the "pylint tests". This code was not verified by Pylint but it was parsed and analyzed so that the test code could be pylint-checked so it's not a full duplication, but I think the savings are significant

Some random stats:

Before the change:

pylint main: 6:20s 
pylint tests: 3:59s 

Tota: 10m 20s

After the change: 
combined pylint: 8:20s


So we save 20% (2 minutes) of elapsed time by doing this.

File: tests/dags/test_logging_in_dag.py (hunk 1 above)
Comment (potiuk) [reply tests/dags/test_logging_in_dag.py:32]: Yep. I had to fix it. Pylint is not 100% deterministic and if you analyze it all together it detects some errors that were not reported before.

**Comment by potiuk** [comment]
I also added yet another small speedup - I realised that when we split pre-commits we should have two separate pre-comit virtualenv caches - that should give another 30-40 seconds boost overall.

**Review by turbaszek — APPROVED** [review]
